app/src/main/java/com/example/tasky/feature_authentication/data/validation/EmailPatternValidatorImpl.kt

```diff
@@ -0,0 +1,10 @@
+package com.example.tasky.feature_authentication.data.validation
+
+import android.util.Patterns
+import com.example.tasky.feature_authentication.domain.validation.EmailPatternValidator
+
+class EmailPatternValidatorImpl: EmailPatternValidator {
+    override fun isEmailPatternValid(email: String): Boolean {
```

Comment:
Little naming tip: Oftentimes, the class or parameter names already provide enough info to understand what something is for, so in this case `isEmailPatternValid` is duplicated. `isValid` would say the same together with the class name `EmailPatternValidator` and the parameter `email`.
What you did is not wrong or bad though, just wanted to add that 😄